libgalois/src/BuildGraph.cpp

```diff
@@ -1899,12 +1899,14 @@ katana::WritePropertyGraph(
     return graph_ptr.error();
   }
 
-  return WritePropertyGraph(std::move(*graph_ptr.value()), dir);
+  // return WritePropertyGraph(std::move(*graph_ptr.value()), dir);
+  // TODO(amber): figure out whether move semantics are the way to go
+  return WritePropertyGraph(*graph_ptr.value(), dir);
 }
 
 katana::Result<void>
 katana::WritePropertyGraph(
-    katana::PropertyGraph prop_graph, const std::string& dir) {
+    katana::PropertyGraph& prop_graph, const std::string& dir) {
```

Comment:
do we need this pass by reference?

Reply:
The write methods in PropertyGraph redirect to RDG's write methods in tsuba, which are non-const. That's what prevents it.

Reply:
I noticed a while back that `const` PropertyGraphs just don't really work. It would be nice to enable them in general. Along with allowing const versions of other related types.